client/asset/dcr/dcr.go

```diff
@@ -3205,10 +3251,11 @@ func (dcr *ExchangeWallet) withdraw(addr stdaddr.Address, val, feeRate uint64) (
 
 // sendToAddress sends an exact amount to an address. Transaction fees will be
 // in addition to the sent amount, and the output will be the zeroth output.
-// TODO: Just use the sendtoaddress rpc since dcrwallet respects locked utxos.
+// TODO: Just use the sendtoaddress rpc since dcrwallet respects locked utxos!
 func (dcr *ExchangeWallet) sendToAddress(addr stdaddr.Address, amt, feeRate uint64) (*wire.MsgTx, uint64, error) {
+	baseSize := uint32(dexdcr.MsgTxOverhead + dexdcr.P2PKHOutputSize*2) // may be extra if change gets omitted (see signTxAndAddChange)
 	enough := func(sum uint64, size uint32, unspent *compositeUTXO) bool {
-		txFee := uint64(size+unspent.input.Size()) * feeRate
+		txFee := uint64(baseSize+size+unspent.input.Size()) * feeRate
```

Comment:
It's curious that this has evaded tests and has not popped up in the wild. Or has it? Regardless, it's probably worth asking if our approach to testing is insufficient.

Reply:
But regarding testing, I don't think "insufficient" is really the problem so much as very cumbersome testing procedures on account of our product code design, specifically the need to coax certain outcomes by stubbing things out with fake data.  It's hard to do, so we don't get thorough unit tests.

In the case of this `enough` closure inside of the `sendToAddress`, it's a closure so we have to test it _indirectly_, which means actually making the following other methods work: `fund`, `spendableUTXOs`, `tryFund`, `lockFundingCoins`, `sendCoins`, `addInputCoins`, `sendWithReturn`, and probably more.  It's understandable that we didn't have narrow unit tests for _just_ these few lines of code in `sendToAddress`.

Perhaps if we defined `enough` at the package level as `enoughForRegularSend` we'd write short and direct unit test without the fuss of stubbing out `dcr.Wallet` and hacking in responses to get the desired effect.

That is all to say I think the general solution is making the bits and pieces more testable on their own, and doing less high level testing of methods that call dozens of other methods.  Integration tests are important, but our unit tests are written like integration tests and fairly difficult to maintain.